extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionConfiguration.java

```diff
@@ -0,0 +1,18 @@
+package io.quarkus.narayana.jta.runtime;
+
+import java.lang.annotation.*;
+
+/**
+ * This annotation can be used to configure a different transaction timeout that the default one for a method or a class.
+ */
+@Inherited
+@Target({ ElementType.METHOD, ElementType.TYPE })
+@Retention(value = RetentionPolicy.RUNTIME)
+public @interface TransactionConfiguration {
+    /**
+     * The transaction timeout in second.
+     * 
+     * @return The transaction timeout in second.
+     */
+    int timeout();
```

Comment:
Make it a default with a specific behavior for the default (maybe zero). Because once we add more settings, one does not have to override all of them.

Reply:
Good catch, I'll use the same value as what was set on the global transaction timeout, 60 seconds, so it's explicit and if we change the transaction implementation it will be backward compatible.

Reply:
done

Reply:
I don't think 60 is a good idea. If only because it can change.

From the TM API, they say that 0 is equivalent to defaulting to the global setting. Better use that.

Reply:
It's to be coherent with what was done on https://github.com/quarkusio/quarkus/pull/2984 we discuss the default to not be 0 (transaction manager default) but 60 to be explicit and allow changing the transaction manager but keepting the default behaviour the same.

Reply:
It was fine to document the global default overriding as it will be used in the doc (automatically generated). But here we do say that we _defer_ to the default / global value.
Let me give you an example, what if I change the quarkus global default to be 120s and I sue this annotation for the isolation level. Will you set the value to 60s because after all that's the hard coded default? Or will you assume that 60 was not set by the user and thus leave it to 120? Remember, you cannot detect in Java that an annotation attribute has been set vs left to the default. Even if, that weird 60s "null value" woudl be very confusing.
Let's use `0` which can be easily documented in the JavaDoc and generally understood as a possible default value by Java devs.

Reply:
Hi, I reply to quickly and you also ;)
I was thinking about it a few minutes ago and yes, 0 is the correct default value as it defers to the default transaction manager value that is already set to 60 to be explicit and independent of the implementation. So, I will default it to 0

Reply:
done